src/java/edu/stanford/math/plex4/graph/UndirectedWeightedListGraph.java

```diff
@@ -93,8 +95,10 @@ public boolean containsEdge(int i, int j) {
 	 * @see edu.stanford.math.plex_plus.graph.AbstractGraph#getNumEdges()
 	 */
 	public int getNumEdges() {
-		// TODO: complete
-		throw new UnsupportedOperationException();
+		int sum = 0;
+		for(int k : degrees.getValues())
+			sum += k;
+		return(sum/2);
```

Comment:
Fix me, if I'm wrong but this would not work if graph made of two adjacent vertices and one self-contained vertex.

```java
addEdge(1, 1);
addEdge(1, 2);
addEdge(3, 3);

// this.digrees will be like as following:
1 => 3
2 => 1
3 => 2
```

There is only one edge, not three.